frontend/src/lib/uPlotLib/utils/getYAxisScale.ts

```diff
@@ -71,23 +72,87 @@ function areAllSeriesEmpty(series: QueryDataV3[]): boolean {
 	});
 }
 
-export const getYAxisScale = (
-	thresholds?: ThresholdProps[],
-	series?: QueryDataV3[],
-	yAxisUnit?: string,
-): {
-	auto: boolean;
-	range?: [number, number];
-} => {
-	if (!thresholds || !series || thresholds.length === 0) return { auto: true };
+function configSoftMinMax(
```

Comment:
@Rajat-Dabade: Please make sure we have a proper test plan for this which can be automated. This will affect all our graphs.

Reply:
Done, added unit test.